resin/logs.py

```diff
@@ -1,133 +1,188 @@
-
-        Examples:
-            # Define callback and error.
-            >>> def callback(message, channel):
-            ...     print(message)
-            >>> def error(message):
-            ...     print('Error:'+ str(message))
-            >>> Logs.subscribe(uuid=uuid, callback=callback, error=error)
-
-        """
+            callback (function): this callback is called on receiving a message.
+            error (Optional[function]): this callback is called on an error event.
+            count (Optional[int]): number of historical messages to include.
 
-        _print_deprecation_warning()
-        channel = self.get_channel(uuid)
-        self.pubnub.subscribe(channels=channel, callback=callback, error=error)
+        Returns:
+            dict: a log entry will contain the following keys: `isStdErr, timestamp, message, isSystem, createdAt`.
 
-    @_init_pubnub
-    def history(self, uuid, callback, error):
         """
-        This function allows fetching historical device logs.
-
-        Args:
-            uuid (str): device uuid.
-            callback (function): this callback is called on receiving a message from the channel.
-            error (function): this callback is called on an error event.
-            For more details about callbacks in pubnub subscribe, visit here: https://www.pubnub.com/docs/python/api-reference#history
-
-        Examples:
-            # Define callback and error.
-            >>> def callback(message):
-            ...     print(message)
-            >>> def error(message):
-            ...     print('Error:'+ str(message))
-            Logs.history(uuid=uuid, callback=callback, error=error)
 
-        """
+        self.device.get(uuid)
+        if uuid is not None and uuid not in self.subscriptions:
+            self.subscriptions[uuid] = self.subscription_handler.add(uuid, callback, error, count)
 
-        _print_deprecation_warning()
-        channel = self.get_channel(uuid)
-        self.pubnub.history(channel=channel, callback=callback, error=error)
+        self.subscription_handler.run()
 
-    def unsubscribe(self, uuid):
+    def history(self, uuid, count=None):
         """
-        This function allows unsubscribing to device logs.
+        Get device logs history.
 
         Args:
             uuid (str): device uuid.
+            count (Optional[int]): this callback is called on receiving a message from the channel.
 
         """
 
-        _print_deprecation_warning()
-        if hasattr(self, 'pubnub'):
-            channel = self.get_channel(uuid)
-            self.pubnub.unsubscribe(channel=channel)
+        raw_query = ''
+
+        if count:
+            raw_query = 'count={}'.format(count)
 
-    def get_channel(self, uuid):
+        return self.base_request.request(
+            '/device/v2/{uuid}/logs'.format(uuid=uuid), 'GET', raw_query=raw_query,
+            endpoint=self.settings.get('api_endpoint')
+        )
+
+    def unsubscribe(self, uuid=None):
```

Comment:
uuid should be mandatory when unsubscribing. if you want to add this functionality I suggest adding an `unsubscribe_all` method

Reply:
yeah, it makes sense to have an `unsubscribe_all` method.